tools/data-handler/src/calculate.ts

```diff
@@ -106,16 +108,27 @@ fieldtype(X, Field, "cardkeys") :- field(X, Field, _), card(Value) : field(X, Fi
             const cardLogicFile = `${filename}.lp`;
             promiseContainer.push(writeFile(cardLogicFile, logicProgram, { encoding: 'utf-8', flag: 'w' }));
         }
+        await Promise.all(promiseContainer);
+    }
 
-        // Once card specific files have been done, write the cardtree.lp.
-        const allCards = parentCard ? await Calculate.project.cards() : cards;
-        let cardTreeContent: string = '';
-        for (const card of allCards) {
-            cardTreeContent += `#include "cards/${card.key}.lp".\n`;
+    // Once card specific files have been done, write the cardtree.lp.
+    private async genereteCardTree() {
+        const destinationFile = join(Calculate.project.calculationFolder, Calculate.cardTreeFileName);
+        const destinationFileBase = join(Calculate.project.calculationFolder, 'cards');
+
+        // Helper to remove extension from filename.
+        function removeExtension(dirent: Dirent) {
+            const name = dirent.name;
+            const index = name.lastIndexOf(".");
+            return (index === -1) ? name : name.substring(0, index);
         }
-        promiseContainer.push(writeFile(destinationFile, cardTreeContent, { encoding: 'utf-8', flag: 'w' }));
 
-        await Promise.all(promiseContainer);
+        const files = await opendir(destinationFileBase);
+        let cardTreeContent: string = '';
+        for await (const file of files) {
```

Comment:
I think the here is `await` is a leftover from some previous implementation that had parallel writing. I will remove it.